lib/modules/datasource/npm/yarnrc.ts

```diff
@@ -0,0 +1,54 @@
+import { load } from 'js-yaml';
+import { z } from 'zod';
+import { logger } from '../../../logger';
+
+export interface YarnConfig {
+  npmRegistryServer?: string;
+  npmScopes?: Record<
+    string,
+    {
+      npmRegistryServer?: string;
+    }
+  >;
+}
+
+const YarnrcYmlSchema = z.object({
+  npmRegistryServer: z.string().optional(),
+  npmScopes: z
+    .record(
+      z.object({
+        npmRegistryServer: z.string().optional(),
+      })
+    )
+    .optional(),
+});
```

Comment:
Keeping both `YarnConfig` and `YarnrcYmlSchema` here so that it's easy to introduce support for legacy Yarn config afterwards

Reply:
We should infer `YarnConfig` from  `YarnrcYmlSchema`